admin/RemoveEmptyAccounts.pl

```diff
@@ -0,0 +1,238 @@
+                    FROM area_tag_raw
+                   WHERE area_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM artist_tag_raw
+                   WHERE artist_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM event_tag_raw
+                   WHERE event_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM instrument_tag_raw
+                   WHERE instrument_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM label_tag_raw
+                   WHERE label_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM place_tag_raw
+                   WHERE place_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM recording_tag_raw
+                   WHERE recording_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_tag_raw
+                   WHERE release_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_group_tag_raw
+                   WHERE release_group_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM series_tag_raw
+                   WHERE series_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM work_tag_raw
+                   WHERE work_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM artist_rating_raw
+                   WHERE artist_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM event_rating_raw
+                   WHERE event_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM label_rating_raw
+                   WHERE label_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM place_rating_raw
+                   WHERE place_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM recording_rating_raw
+                   WHERE recording_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_group_rating_raw
+                   WHERE release_group_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM work_rating_raw
+                   WHERE work_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM annotation
+                   WHERE annotation.editor = e.id)
+  AND NOT EXISTS (   SELECT 1
+                       FROM editor_collection ec
+                      WHERE ec.editor = e.id
+                        AND ((
+                                  (type != 5 OR name != 'Attending')
+                                  AND
+                                  (type != 6 OR name != 'Maybe attending')
+                            )
+                            OR EXISTS (SELECT 1
+                                         FROM editor_collection_event ece
+                                        WHERE ece.collection = ec.id)))
```

Comment:
I'm finding this hard to understand, but also wondering: was there a reason we don't just check if all existing collections are empty? Looks like it's trying to detect the presence of default collections, but I don't remember why that was necessary.

Reply:
IIRC the idea was that we wanted to avoid autoremoving editors who had created a collection themselves, even if empty. Although I'd agree it's fine to remove them if they added an empty collection over a year ago tbh.